eng/copy-wpf.ps1

```diff
@@ -0,0 +1,126 @@
+}
+
+function CopyThemeBinariesToPublishLocation($publishLocation, $localBinLocation)
+{
+    $ThemeBinaries = "PresentationFramework.Aero", "PresentationFramework.Aero2", "PresentationFramework.AeroLight", "PresentationFramework.Classic", "PresentationFramework.Luna", "PresentationFramework.Royale"
+    CopyNetCoreApp3Binaries $publishLocation $localBinLocation $ThemeBinaries
+}
+
+function CopyUIAutomationBinariesToPublishLocation($publishLocation, $localBinLocation)
+{
+    $UIAutomationBinaries = "UIAutomationClient", "UIAutomationClientSideProviders", "UIAutomationProvider", "UIAutomationTypes"
+    CopyNetCoreApp3Binaries $publishLocation $localBinLocation $UIAutomationBinaries
+}
+
+function CopyNetCoreApp3Binaries($publishLocation, $localBinLocation, $binaries)
+{
+    $ArchFolder = if ($arch -eq "x86") { "" } else { "x64" }
+    foreach($binary in $binaries)
+    {
+        $FullBinaryName = $binary + ".dll"
+        $BinLocation = Join-Path (Join-Path (Join-Path (Join-Path $localBinLocation $binary) $ArchFolder) $Config) "netcoreapp3.0"
+        CopyBinaryToLocation $FullBinaryName $BinLocation $publishLocation
+    }
+}
+
+function CopyBinaryToLocation($binary, $binaryLocation, $destination)
+{
+    $FullBinaryPath = Join-Path $binaryLocation $binary
+    if (Test-Path $FullBinaryPath)
+    {
+        Write-Host "Copying " $binary " to " $destination  
+        Copy-Item $FullBinaryPath $destination -Force
+    }
+    else
+    {
+        Write-Host "Binary Unavailable: " $binary -ForegroundColor Yellow
+    }
+}
+
+function Print-Usage()
+{
+    Write-Host "Usage: copy-wpf.ps1 -destination <value> [-arch <value] [-release] [-shared]"
+    Write-Host "    This script helps developers deploy wpf assemblies to the proper location for easy testing. See "
+    Write-Host "    developer-guide.md for more information on how to use this script."
+    Write-Host ""
+    Write-Host "Common parameters:"
+    Write-Host "  -destination <value>    Location of .csproj or .vbproj of application to test against. If copying"
+    Write-Host "                          over a shared installation, the value should be the version to copy over."
+    Write-Host "  -arch <value>           Architecture of binaries to copy. Can be either x64 or x86. Default is x86."
+    Write-Host "  -release                Copy release binaries. Default is to copy Debug binaries"
+    Write-Host "  -shared                 Copy binaries over the shared installation of Microsoft.WindowsDesktop.App."
+    Write-Host "  -help                   Print help and exit"
+    Write-Host ""
+}
+
+$RepoRoot = Resolve-Path (Join-Path $PSScriptRoot "..")
+$Config = if ($release) { "Release" } else { "Debug" }
+
+if ($help -or [string]::IsNullOrEmpty($destination)) 
+{
+    Print-Usage
+}
+elseif($shared)
+{
+    Write-Host "Copying binaries to shared installation"
+    $sharedLocation = Join-Path (Join-Path $env:ProgramFiles "dotnet\shared\Microsoft.WindowsDesktop.App") $destination
```

Comment:
A shared installed can be somewhere besides %programfiles%.

Reply:
yeah you are right, although i'm not sure how we could figure out where? regardless, i think after the dicussion we had yesterday i'm going to remove this in favor of doing a test-host installation now that i have a better understanding of how that is done.